proto/osmosis/superfluid/tx.proto

```diff
@@ -231,4 +236,35 @@ message MsgAddToConcentratedLiquiditySuperfluidPositionResponse {
     (gogoproto.nullable) = false
   ];
   uint64 lock_id = 4 [ (gogoproto.moretags) = "yaml:\"lock_id\"" ];
+}
+
+// ===================== MsgUnbondConvertAndStake
+message MsgUnbondConvertAndStake {
+  option (amino.name) = "osmosis/unbond-convert-and-stake";
+
+  uint64 lock_id = 1 [ (gogoproto.moretags) = "yaml:\"lock_id\"" ];
+  string sender = 2 [ (gogoproto.moretags) = "yaml:\"sender\"" ];
+  string val_addr = 3;
```

Comment:
We should just use the valset preference module.

Reply:
What if person who's staking does not have val set pref set?

Reply:
Then the tx should just fail

Reply:
(if they have any stake already, then valset prefs module, fallsback to their current distribution.  so it will only fail if they don't have any staked OSMO already AND they haven't set a preference).

Reply:
So is the FE supposed to set valset pref + migrate?

If valset pref is unset, and you've delegated to too many validators, it may also fail

Reply:
proposal: how about we have the val address field as fall back where it would be used as fallback when staking through valset failed?

Reply:
mm okay. As long as that's documented.  In the frontend, we will probably not support that tho, to keep it simple

Reply:
Let's document this decision in this PR please

Reply:
Changed in https://github.com/osmosis-labs/osmosis/pull/5949/commits/0898bdc25c91d561e5f85dac87b4e3dca69431f4